### DISCUSSION TITLE
Installing PHP 8 using Homebrew issue

### DISCUSSION THREAD

**Discussion by WintersDoor** [Homebrew/discussions#241] [Getting started] [answered]
Hello,

I wrote this on the Discourse version, but I wanted to see if I can get replies here too. Anyways, I know that PHP 8 just came out a few weeks ago, but I'm having problems installing it. I used to use Liip's PHP but am forced to run PHP via Homebrew since Liip deprecated their PHP binary files. I'm able to install PHP 8 CLI using Homebrew, but when trying to load the PHP module for Apache, I keep getting this error.

```
Cannot load /usr/local/opt/php/lib/httpd/modules/libphp.so into server: dlopen(/usr/local/opt/php/lib/httpd/modules/libphp.so, 10): Symbol not found: _curl_mime_addpart\n  Referenced from: /usr/local/opt/php/lib/httpd/modules/libphp.so\n  Expected in: /usr/lib/libcurl.4.dylib\n in /usr/local/opt/php/lib/httpd/modules/libphp.so
```

I know it has to do something with the cURL extension not being enabled. But I can't seem to figure out how to do this. Do I have to reinstall cURL as a whole?

**Comment by gromgit** [comment, marked as answer]
Are you using the system Apache httpd? As far as I know, Homebrew PHP only supports Homebrew httpd.

**Reply by WintersDoor** [reply]
I originally was using the default Apache when I was using Liip. But I assume I’m not anymore because I used this command to unload the default Apache.

`sudo launchctl unload -w /System/Library/LaunchDaemons/org.apache.httpd.plist`

So as far as I know, I think I’m not using the default anymore.

**Reply by SMillerDev** [reply]
If you don't know what apache you're using, where an how does this error show up?

**Reply by WintersDoor** [reply]
Well I was a bit confused on why httpd from Homebrew wasn't loading up. So I did a few tests. Until now, I didn't know what was really happening because I unloaded the default Apache and tried to start up httpd from Homebrew. Once I refreshed my Firefox tab using `localhost/info.php`, I got

```
Unable to connect

Firefox can’t establish a connection to the server at localhost.

The site could be temporarily unavailable or too busy. Try again in a few moments.
If you are unable to load any pages, check your computer’s network connection.
If your computer or network is protected by a firewall or proxy, make sure that Firefox is permitted to access the Web.
```

So Homebrew's httpd seems to be crashing even though in the Terminal, there's nothing wrong. Then I tried to restart the default up and refreshed the same tab. My localhost came back up and displayed my PHP version which was installed using Liip. So that means that the PHP version from Homebrew wasn't loaded in. That's where I did `sudo /usr/sbin/apachectl configtest` and that's where I got that error from.

Then I decided since my localhost keeps loading up the default Apache, I would just copy and paste over the httpd.conf file from Homebrew. That way, it'll target the Homebrew files using the default Apache's httpd.conf file. So I tried that and I still get the same error message. So I came to the assumption that it has something to do with `libphp.so` having some issue with cURL.

**Reply by gromgit** [reply]
> `sudo launchctl unload -w /System/Library/LaunchDaemons/org.apache.httpd.plist`

IIRC, that just disables and unloads the stock http config, and doesn't stop any stock httpd processes that are already running. To confirm:
```
ps -ef | grep httpd
```

If the stock httpd is still running, this would explain:

> So Homebrew's httpd seems to be crashing even though in the Terminal, there's nothing wrong.

Since your Homebrew httpd is using the same config file as the stock httpd, it's crashing precisely because the stock httpd is still running, and therefore hogging the port(s) it needs.

**Reply by WintersDoor** [reply]
Got it. Thanks. That helped a lot. The default Apache was still running so I did a `sudo /usr/sbin/apachectl stop`. Once I did that, I had to search for Homebrew's `apachectl` file which was located in `/usr/local/Cellar/httpd/2.4.46/bin/`. Then I did a `sudo /usr/local/Cellar/httpd/2.4.46/bin/apachectl start` and then refreshed my tab and it worked. Then did `ps -ef | grep httpd` just to confirm that Homebrew's `apachectl` is only running and it was.

**Reply by gromgit** [reply]
> I had to search for Homebrew's `apachectl` file

Or you could've just used `/usr/local/bin/apachectl` instead, which should always point to the current binary.

**Comment by carlocab** [comment]
Please include exact reproduction steps. It's very hard for other people to help you if they don't know how to reproduce your error.
